modules/hotmaps/1.0/hotmaps.smk

```diff
@@ -0,0 +1,837 @@
+    output:
+        dnps = temp(CFG["dirs"]["inputs"] + "maf/{sample_set}--{launch_date}/{md5sum}.dnps.maf"),
+        completed = CFG["dirs"]["inputs"] + "maf/{sample_set}--{launch_date}/{md5sum}_split_dnps.complete"
+    shell:
+        op.as_one_line("""
+        variant_type_col=$(head -n 1 {input.maf} | sed 's/\\t/\\n/g' | nl | grep "Variant_Type" | cut -f 1) &&
+        protein_position_col=$(head -n 1 {input.maf} | sed 's/\\t/\\n/g' | nl | grep "Protein_position" | cut -f 1) &&
+        cat <( head -n 1 {input.maf} ) <( awk -v var_col="$variant_type_col" -v protein_col="$protein_position_col" ' {{ if ( $var_col=="DNP" && $protein_col ~ /[0-9?]+[-][0-9?]+/) print $0 }} ' {input.maf} ) > {output.dnps} &&
+        touch {output.completed}
+        """)
+
+checkpoint _hotmaps_maf2vcf:
+    input:
+        dnps = str(rules._hotmaps_split_dnps.output.completed),
+        fasta = reference_files("genomes/grch37/genome_fasta/genome.fa")
+    output:
+        done = CFG["dirs"]["maf2vcf"] + "completed/{sample_set}--{launch_date}/{md5sum}.done"
+    params:
+        vcf = CFG["dirs"]["maf2vcf"] + "vcf/{sample_set}--{launch_date}/{md5sum}.dnps.vcf",
+        vcf_dir = CFG["dirs"]["maf2vcf"] + "vcf/{sample_set}--{launch_date}/{md5sum}/",
+        dnps = str(rules._hotmaps_split_dnps.output.dnps)
+    conda:
+        CFG["conda_envs"]["bcftools"]
+    log:
+        stdout = CFG["logs"]["maf2vcf"] + "{sample_set}--{launch_date}/{md5sum}_maf2vcf.stdout.log",
+        stderr = CFG["logs"]["maf2vcf"] + "{sample_set}--{launch_date}/{md5sum}_maf2vcf.stderr.log"
+    shell:
+        op.as_one_line("""
+        if [ $( wc -l < {params.dnps}) -gt 1 ] ;
+        then
+            mkdir -p {params.vcf_dir} &&
+            maf2vcf.pl 
+            --input-maf {params.dnps} 
+            --output-dir {params.vcf_dir} 
+            --output-vcf {params.vcf} 
+            --ref-fasta {input.fasta} 
+            --per-tn-vcfs 
+            > {log.stdout} 2> {log.stderr} ;
+        fi &&
+        touch {output.done}
+        """)
+
+rule _hotmaps_bcftools:
+    input:
+        vcf = CFG["dirs"]["maf2vcf"] + "vcf/{sample_set}--{launch_date}/{md5sum}/{tumour_id}_vs_{normal_sample_id}.vcf",
+        dnp_vcf = str(rules._hotmaps_maf2vcf.output.done)
+    output:
+        vcf = CFG["dirs"]["bcftools"] + "vcf/{sample_set}--{launch_date}/{md5sum}/{tumour_id}_vs_{normal_sample_id}.annotate.vcf"
+    conda:
+        CFG["conda_envs"]["bcftools"] 
+    log:
+        stdout = CFG["logs"]["bcftools"] + "{sample_set}--{launch_date}/{md5sum}/{tumour_id}_vs_{normal_sample_id}/bcftools_norm.stdout.log",
+        stderr = CFG["logs"]["bcftools"] + "{sample_set}--{launch_date}/{md5sum}/{tumour_id}_vs_{normal_sample_id}/bcftools_norm.stderr.log"
+    shell:
+        op.as_one_line("""
+        bcftools norm --atomize 
+        --output {output.vcf} 
+        --output-type v {input.vcf} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+rule _hotmaps_vcf2maf:
```

Comment:
so basically you are running maf to vcf and then back to maf just to get rid of DNPs?
The potential issue with this is not sure how the alternative transcripts are handled. In gambl the "standard" vcf2maf uses a set of custom ENST because they are incorrectly annotated by vcf2maf and there are some important genes like MYD88, FOXO1 etc. Is that getting lost here after another pass to vcf2maf?

Reply:
Hmm I didn't realize -- I'll look into this

Reply:
I looked into the previous results and found that the output protein positions were being changed in DNP variants compared to their previous MAFs! I added the ability to provide a "custom enst" file in the options for the vcf2maf rule, supplied the one available in gambl at `etc/custom_enst.grch37.txt` and this fixed the discordant protein positions